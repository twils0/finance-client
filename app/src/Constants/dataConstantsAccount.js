export const actionTypes = {
  SET_ACCOUNT_STATUS: 'DATA_ACCOUNT_SET_ACCOUNT_STATUS',
  SET_FIELD: 'DATA_ACCOUNT_SET_FIELD',
  SET_FIELDS: 'DATA_ACCOUNT_SET_FIELDS',
  RESET_ACCOUNT_STATE: 'DATA_ACCOUNT_RESET_ACCOUNT_STATE',
};

export const statusNames = {
  AWS_FIELDS: 'accountStatusAWSFields',
  STRIPE_FIELDS: 'accountStatusStripeFields',
  UPDATE_STRIPE_FIELDS_TOKEN: 'accountStatusUpdateStripeFieldsToken',
  UPDATE_STRIPE_FIELDS_REQUEST: 'accountStatusUpdateStripeFieldsRequest',
  UPDATE_DB_FIELDS: 'accountStatusUpdateDBFields',
  UPDATE_AWS_FIELDS: 'accountStatusUpdateAWSFields',
};

export const fieldNames = {
  NAME: 'accountName',
  EMAIL: 'accountEmail',
  EMAIL_ADDITIONAL: 'accountEmailAdditional',
  PHONE: 'accountPhone',
  NAME_ON_CARD: 'accountNameOnCard',
  PROMO_CODE: 'accountPromoCode',
  PROMO_CODE_VALID: 'accountPromoCodeValid',
  CUSTOMER: 'accountCustomer',
};
