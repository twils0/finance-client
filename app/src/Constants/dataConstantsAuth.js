export const actionTypes = {
  SET_AUTHENTICATED: 'DATA_AUTH_SET_AUTHENTICATED',
  SET_REDIRECT_URL: 'DATA_AUTH_SET_REDIRECT_URL',
  SET_AUTH_STATUS: 'DATA_AUTH_SET_AUTH_STATUS',
  SET_CODE_TYPE: 'DATA_AUTH_SET_CODE_TYPE',
  RESET_AUTH_STATE: 'DATA_AUTH_RESET_AUTH_STATE',
};

export const statusNames = {
  LOGIN: 'authStatusLogin',
  SIGN_OUT_DEVICES: 'authStatusSignOutDevices',
  LOGIN_MFA: 'authStatusLoginMFA',
  SIGN_UP: 'authStatusSignUp',
  VERIFY_PHONE: 'authStatusVerifyPhone',
  VERIFY_PHONE_CODE: 'authStatusVerifyPhoneCode',
  VERIFY_EMAIL: 'authStatusVerifyEmail',
  VERIFY_EMAIL_LINK: 'authStatusVerifyEmailLink',
  FORGOT_PASSWORD: 'authStatusForgotPassword',
  RESET_PASSWORD: 'authStatusResetPassword',
  CHANGE_PASSWORD: 'authStatusChangePassword',
  DELETE_ACCOUNT: 'authStatusDeleteAccount',
  LOGOUT: 'authStatusLogout',
};

export const codeTypeNames = {
  VERIFY_PHONE: 'verifyPhone',
  VERIFY_EMAIL: 'verifyEmail',
  VERIFY_EMAIL_ADDITIONAL: 'verifyEmailAdditional',
};
