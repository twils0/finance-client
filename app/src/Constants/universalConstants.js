import stripeConfig from '../../../stripe.config.json';
import dbConfig from '../../../db.config.json';
import iexConfig from '../../../iex.config.json';

export const baseURLs = {
  STRIPE: stripeConfig.url,
  DB: dbConfig.url,
  IEX: iexConfig.urls.api,
  IEX_DEV: iexConfig.urls.dev,
  IEX_TERMS: iexConfig.urls.terms,
};

export const URLs = {
  CUSTOMERS: '/customers',
  USERS: '/users',
  EMAILS: '/emails',
  SECURITIES: '/securities',
  SECURITIES_SEARCH: '/securities/search',
  IEX_COMPANY: tickerCusip => `/stock/${tickerCusip}/company`,
  IEX_QUOTE: tickerCusip => `/stock/${tickerCusip}/quote`,
};

export const axiosConfig = {
  STRIPE: {
    baseURL: baseURLs.STRIPE,
    timeout: 30000,
  },
  DB: {
    baseURL: baseURLs.DB,
    timeout: 30000,
  },
  IEX: {
    baseURL: baseURLs.IEX,
    timeout: 30000,
  },
};

export const pathNames = {
  DEFAULT: '*',
  LOGIN: '/login',
  SIGN_UP: '/signup',
  CODE: '/code',
  CODE_VERIFY_EMAIL: '/code/:verificationId?',
  TERMS: '/terms',
  INTERNAL: '/internal',
  WATCHLIST: '/internal/watchlist',
  WATCHLIST_SECURITY_ID: '/internal/watchlist/:securityId?',
  ACCOUNT: '/internal/account',
};

export const requestStatusTypes = {
  IDLE: 'IDLE',
  LOADING: 'LOADING',
  SUCCESS: 'SUCCESS',
  ERROR: 'ERROR',
};
