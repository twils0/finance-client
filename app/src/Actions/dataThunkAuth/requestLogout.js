import { Auth } from 'aws-amplify';
import raven from 'raven-js';

import handleErrorCatch from '../../handleErrorCatch';
import { requestStatusTypes } from '../../Constants/universalConstants';
import { statusNames } from '../../Constants/dataConstantsAuth';
import { formNames, buttonNames, buttonTexts } from '../../Constants/uiConstantsAccount';
import { setAuthStatus, resetAuthState } from '../dataActionsAuth';
import { resetAWSState } from '../dataActionsAWS';
import { resetAccountState } from '../dataActionsAccount';
import { resetWatchlistState } from '../dataActionsWatchlist';
import { setCurrentForm, setButtonText, setButtonVisible } from '../uiActionsAccount';

const requestLogout = () => async (dispatch, getState) => {
  const state = getState();

  if (state.data.auth.status[statusNames.LOGOUT].status !== requestStatusTypes.LOADING) {
    dispatch(
      setAuthStatus({
        id: statusNames.LOGOUT,
        status: requestStatusTypes.LOADING,
      }),
    );

    if (state.ui.internal.account.forms.current !== formNames.PROFILE) {
      dispatch(setCurrentForm({ current: formNames.PROFILE }));
    }

    if (state.ui.internal.account.buttons[buttonNames.FIRST].text !== buttonTexts.EDIT) {
      dispatch(setButtonText({ id: buttonNames.FIRST, text: buttonTexts.EDIT }));
    }

    if (state.ui.internal.account.buttons[buttonNames.SECOND].visible) {
      dispatch(setButtonVisible({ id: buttonNames.SECOND, visible: false }));
    }

    try {
      await Auth.signOut();
    } catch (errorCatch) {
      const error = handleErrorCatch(errorCatch);

      dispatch(
        setAuthStatus({
          id: statusNames.LOGOUT,
          status: requestStatusTypes.ERROR,
        }),
      );

      raven.captureException(error, {
        logger: 'requestLogout',
      });

      return Promise.reject(error);
    }

    window.sessionStorage.removeSecurity('sessionTime');

    const intervalId = window.sessionStorage.getSecurity('interval');

    if (intervalId) {
      window.clearInterval(intervalId);

      window.sessionStorage.removeSecurity('interval');
    }

    dispatch(resetAWSState());
    dispatch(resetAccountState());
    dispatch(resetWatchlistState());
    dispatch(resetAuthState());

    raven.setUserContext();
  }

  return null;
};

export default requestLogout;
