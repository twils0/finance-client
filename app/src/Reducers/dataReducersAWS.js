import { combineReducers } from 'redux';

import { requestStatusTypes } from '../Constants/universalConstants';
import { actionTypes } from '../Constants/dataConstantsAWS';

export const initialState = {
  status: requestStatusTypes.IDLE,
  user: null,
};

export const status = (state = initialState.status, action) => {
  switch (action.type) {
    case actionTypes.SET_AWS_STATUS:
      return action.payload.status;
    case actionTypes.RESET_AWS_STATE:
      return initialState.status;
    default:
      return state;
  }
};

export const user = (state = initialState.user, action) => {
  switch (action.type) {
    case actionTypes.SET_AWS_USER:
      return action.payload.user;
    case actionTypes.RESET_AWS_STATE:
      return initialState.user;
    default:
      return state;
  }
};

const dataReducersAWS = combineReducers({
  status,
  user,
});

export default dataReducersAWS;
