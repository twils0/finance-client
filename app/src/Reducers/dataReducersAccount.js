import { combineReducers } from 'redux';

import { requestStatusTypes } from '../Constants/universalConstants';
import { actionTypes, statusNames, fieldNames } from '../Constants/dataConstantsAccount';

export const initialState = {
  status: {
    [statusNames.AWS_FIELDS]: {
      id: statusNames.AWS_FIELDS,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.STRIPE_FIELDS]: {
      id: statusNames.STRIPE_FIELDS,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.UPDATE_STRIPE_FIELDS_TOKEN]: {
      id: statusNames.UPDATE_STRIPE_FIELDS_TOKEN,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.UPDATE_STRIPE_FIELDS_REQUEST]: {
      id: statusNames.UPDATE_STRIPE_FIELDS_REQUEST,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.UPDATE_DB_FIELDS]: {
      id: statusNames.UPDATE_DB_FIELDS,
      status: requestStatusTypes.IDLE,
    },
    [statusNames.UPDATE_AWS_FIELDS]: {
      id: statusNames.UPDATE_AWS_FIELDS,
      status: requestStatusTypes.IDLE,
    },
  },
  fields: {
    list: [...Object.values(fieldNames)],
    [fieldNames.NAME]: {
      id: fieldNames.NAME,
      value: '',
    },
    [fieldNames.EMAIL]: {
      id: fieldNames.EMAIL,
      value: '',
    },
    [fieldNames.EMAIL_ADDITIONAL]: {
      id: fieldNames.EMAIL_ADDITIONAL,
      value: '',
    },
    [fieldNames.PHONE]: {
      id: fieldNames.PHONE,
      value: '',
    },
    [fieldNames.NAME_ON_CARD]: {
      id: fieldNames.NAME_ON_CARD,
      value: '',
    },
    [fieldNames.PROMO_CODE]: {
      id: fieldNames.PROMO_CODE,
      value: '',
    },
    [fieldNames.PROMO_CODE_VALID]: {
      id: fieldNames.PROMO_CODE_VALID,
      value: true,
    },
    [fieldNames.CUSTOMER]: {
      id: fieldNames.CUSTOMER,
      value: '',
    },
  },
};

export const status = (state = initialState.status, action) => {
  switch (action.type) {
    case actionTypes.SET_ACCOUNT_STATUS:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.RESET_ACCOUNT_STATE:
      return initialState.status;
    default:
      return state;
  }
};

export const fields = (state = initialState.fields, action) => {
  switch (action.type) {
    case actionTypes.SET_FIELD:
      return {
        ...state,
        [action.payload.id]: {
          ...state[action.payload.id],
          ...action.payload,
        },
      };
    case actionTypes.SET_FIELDS:
      return {
        ...state,
        ...action.payload,
      };
    case actionTypes.RESET_ACCOUNT_STATE:
      return initialState.fields;
    default:
      return state;
  }
};

const dataReducersAccount = combineReducers({
  status,
  fields,
});

export default dataReducersAccount;
