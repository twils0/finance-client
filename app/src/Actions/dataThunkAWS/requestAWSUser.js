import { Auth } from 'aws-amplify';
import raven from 'raven-js';

import handleErrorCatch from '../../handleErrorCatch';
import { requestStatusTypes } from '../../Constants/universalConstants';
import { setAWSStatus, setAWSUser } from '../dataActionsAWS';
import requestLogout from '../dataThunkAuth/requestLogout';

import { demo } from '../../../../mode.config.json';

const requestAWSUser = () => async (dispatch, getState) => {
  const state = getState();

  if (state.data.aws.status !== requestStatusTypes.LOADING) {
    dispatch(setAWSStatus({ status: requestStatusTypes.LOADING }));

    let user = null;

    try {
      if (!demo) {
        user = await Auth.currentUserPoolUser();
      }
    } catch (errorCatch) {
      const error = handleErrorCatch(errorCatch);

      dispatch(setAWSStatus({ status: requestStatusTypes.ERROR }));

      if (
        (typeof error === 'string'
          && (error === 'not authenticated' || error === 'No current user in userPool'))
        || (typeof error === 'object' && error && error.code === 'NotAuthorizedException')
      ) {
        dispatch(requestLogout());

        return null;
      }

      if (!demo) {
        raven.captureException(error, {
          logger: 'requestAWSUser',
        });
      }

      return Promise.reject(error);
    }

    if (!demo) {
      dispatch(setAWSUser({ user }));
    }
    dispatch(setAWSStatus({ status: requestStatusTypes.SUCCESS }));

    return { user };
  }

  return null;
};

export default requestAWSUser;
