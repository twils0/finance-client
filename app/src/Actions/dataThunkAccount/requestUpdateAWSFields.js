import { Auth } from 'aws-amplify';
import raven from 'raven-js';

import { requestStatusTypes } from '../../Constants/universalConstants';
import { statusNames } from '../../Constants/dataConstantsAccount';
import requestLogout from '../dataThunkAuth/requestLogout';
import requestAWSUser from '../dataThunkAWS/requestAWSUser';
import { setAccountStatus } from '../dataActionsAccount';

import handleErrorCatch from '../../handleErrorCatch';

const requestUpdateAWSFields = (payload) => {
  if (
    !Object.prototype.hasOwnProperty.call(payload, 'email') &&
    !Object.prototype.hasOwnProperty.call(payload, 'name') &&
    !Object.prototype.hasOwnProperty.call(payload, 'phone')
  ) {
    throw new Error(`Please enter a value for either the 'email', 'name', or 'phone' keys- ${JSON.stringify(payload)}`);
  }

  return async (dispatch, getState) => {
    const state = getState();

    if (
      state.data.account.status[statusNames.UPDATE_AWS_FIELDS].status !== requestStatusTypes.LOADING
    ) {
      dispatch(setAccountStatus({
        id: statusNames.UPDATE_AWS_FIELDS,
        status: requestStatusTypes.LOADING,
      }));

      const updatePayload = { ...payload };
      if (payload.phone) {
        delete updatePayload.phone;
        updatePayload.phone_number = payload.phone;
      }
      if (payload.emailAdditional) {
        delete updatePayload.emailAdditional;
      }

      let { user } = state.data.aws;

      try {
        if (!user) {
          ({ user } = await dispatch(requestAWSUser()));
        }

        await Auth.updateUserAttributes(user, updatePayload);
      } catch (errorCatch) {
        const error = handleErrorCatch(errorCatch);

        dispatch(setAccountStatus({
          id: statusNames.UPDATE_AWS_FIELDS,
          status: requestStatusTypes.ERROR,
        }));

        if (typeof error === 'object' && error.code === 'NotAuthorizedException') {
          dispatch(requestLogout());

          return null;
        }

        if (
          typeof error === 'object' &&
          error &&
          (error.code === 'AliasExistsException' || error.code === 'UsernameExistsException')
        ) {
          return Promise.reject(error);
        }

        raven.captureException(error, {
          logger: 'requestUpdateAWSFields',
        });

        return Promise.reject(error);
      }

      dispatch(setAccountStatus({
        id: statusNames.UPDATE_AWS_FIELDS,
        status: requestStatusTypes.SUCCESS,
      }));
    }

    return null;
  };
};

export default requestUpdateAWSFields;
