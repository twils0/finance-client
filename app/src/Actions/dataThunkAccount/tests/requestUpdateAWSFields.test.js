import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Auth } from 'aws-amplify';

import { initialState as initialStateAWS } from '../../../Reducers/dataReducersAWS';
import { initialState as initialStateAccount } from '../../../Reducers/dataReducersAccount';
import { requestStatusTypes } from '../../../Constants/universalConstants';
import {
  actionTypes as actionTypesAccount,
  statusNames,
} from '../../../Constants/dataConstantsAccount';
import requestLogout from '../../dataThunkAuth/requestLogout';
import requestAWSUser from '../../dataThunkAWS/requestAWSUser';
import requestUpdateAWSFields from '../requestUpdateAWSFields';

jest.mock('aws-amplify', () => ({
  Auth: {
    updateUserAttributes: jest.fn(),
  },
}));
jest.mock('../../dataThunkAuth/requestLogout', () => jest.fn());
jest.mock('../../dataThunkAWS/requestAWSUser', () => jest.fn());

const middleware = [thunk];
const mockStore = configureMockStore(middleware);

const userObject = { user: { test: 'testUser' } };
const email = 'test1@test.com';
const name = 'testName';
const phone = 'testPhone';
const updatePayload = {
  email,
  emailAdditional: 'test2@test.com',
  name,
  phone,
};
const expectedUpdatePayload = {
  email,
  name,
  phone_number: phone,
};

describe('dataThunkAccount', () => {
  describe('requestUpdateAWSFields', () => {
    afterEach(() => {
      requestAWSUser.mockReset();
      Auth.updateUserAttributes.mockReset();
      requestLogout.mockReset();
    });

    it("fails and throws an error when missing 'name', 'email', and 'phone_number' keys in payload", async () => {
      const stateBeforeAWS = JSON.parse(JSON.stringify(initialStateAWS));
      stateBeforeAWS.user = userObject.user;

      const store = mockStore({ data: { aws: stateBeforeAWS, account: initialStateAccount } });

      Auth.updateUserAttributes.mockReturnValue(Promise.resolve());

      const emptyPayload = {};

      try {
        await store.dispatch(requestUpdateAWSFields(emptyPayload));
      } catch (errorCatch) {
        expect(errorCatch).toMatchSnapshot();
      }

      expect(requestAWSUser).not.toBeCalled();
      expect(Auth.updateUserAttributes).not.toBeCalled();
      expect(requestLogout).not.toBeCalled();
    });

    it("calls requestAWSUser when missing 'user' object in state", async () => {
      const store = mockStore({ data: { aws: initialStateAWS, account: initialStateAccount } });

      requestAWSUser.mockReturnValue(() => Promise.resolve(userObject));
      Auth.updateUserAttributes.mockReturnValue(Promise.resolve());

      await store.dispatch(requestUpdateAWSFields(updatePayload));

      expect(requestAWSUser).toBeCalled();
      expect(Auth.updateUserAttributes).toBeCalledWith(userObject.user, expectedUpdatePayload);
      expect(requestLogout).not.toBeCalled();
    });

    it('fails and returns promise reject when requestAWSUser throws an error', async () => {
      const store = mockStore({ data: { aws: initialStateAWS, account: initialStateAccount } });

      const error = { code: 'testError', message: 'testMessage' };
      const expectedActions = [
        {
          type: actionTypesAccount.SET_ACCOUNT_STATUS,
          payload: {
            id: statusNames.UPDATE_AWS_FIELDS,
            status: requestStatusTypes.LOADING,
          },
        },
        {
          type: actionTypesAccount.SET_ACCOUNT_STATUS,
          payload: {
            id: statusNames.UPDATE_AWS_FIELDS,
            status: requestStatusTypes.ERROR,
          },
        },
      ];

      requestAWSUser.mockReturnValue(() => Promise.reject(error));

      try {
        await store.dispatch(requestUpdateAWSFields(updatePayload));
      } catch (errorCatch) {
        expect(errorCatch).toEqual(error);
      }

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
      expect(Auth.updateUserAttributes).not.toBeCalled();
      expect(requestLogout).not.toBeCalled();
    });

    it("calls requestLogout when Auth.updateUserAttributes throws 'NotAuthorizedException' error", async () => {
      const stateBeforeAWS = JSON.parse(JSON.stringify(initialStateAWS));
      stateBeforeAWS.user = userObject.user;

      const store = mockStore({ data: { aws: stateBeforeAWS, account: initialStateAccount } });

      const error = { code: 'NotAuthorizedException', message: 'testMessage' };
      const expectedActions = [
        {
          type: actionTypesAccount.SET_ACCOUNT_STATUS,
          payload: {
            id: statusNames.UPDATE_AWS_FIELDS,
            status: requestStatusTypes.LOADING,
          },
        },
        {
          type: actionTypesAccount.SET_ACCOUNT_STATUS,
          payload: {
            id: statusNames.UPDATE_AWS_FIELDS,
            status: requestStatusTypes.ERROR,
          },
        },
      ];

      Auth.updateUserAttributes.mockReturnValue(Promise.reject(error));
      requestLogout.mockReturnValue(() => null);

      try {
        await store.dispatch(requestUpdateAWSFields(updatePayload));
      } catch (errorCatch) {
        expect(errorCatch).toEqual(null);
      }

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
      expect(requestAWSUser).not.toBeCalled();
      expect(Auth.updateUserAttributes).toBeCalledWith(userObject.user, expectedUpdatePayload);
      expect(requestLogout).toBeCalled();
    });

    it('fails and returns promise reject when Auth.updateUserAttributes throws an error', async () => {
      const stateBeforeAWS = JSON.parse(JSON.stringify(initialStateAWS));
      stateBeforeAWS.user = userObject.user;

      const store = mockStore({ data: { aws: stateBeforeAWS, account: initialStateAccount } });

      const error = { code: 'testError', message: 'testMessage' };
      const expectedActions = [
        {
          type: actionTypesAccount.SET_ACCOUNT_STATUS,
          payload: {
            id: statusNames.UPDATE_AWS_FIELDS,
            status: requestStatusTypes.LOADING,
          },
        },
        {
          type: actionTypesAccount.SET_ACCOUNT_STATUS,
          payload: {
            id: statusNames.UPDATE_AWS_FIELDS,
            status: requestStatusTypes.ERROR,
          },
        },
      ];

      Auth.updateUserAttributes.mockReturnValue(Promise.reject(error));

      try {
        await store.dispatch(requestUpdateAWSFields(updatePayload));
      } catch (errorCatch) {
        expect(errorCatch).toEqual(error);
      }

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
      expect(requestAWSUser).not.toBeCalled();
      expect(Auth.updateUserAttributes).toBeCalledWith(userObject.user, expectedUpdatePayload);
      expect(requestLogout).not.toBeCalled();
    });

    it('creates the correct actions with the correct payloads', async () => {
      const stateBeforeAWS = JSON.parse(JSON.stringify(initialStateAWS));

      stateBeforeAWS.user = userObject.user;
      const store = mockStore({ data: { aws: stateBeforeAWS, account: initialStateAccount } });

      const expectedActions = [
        {
          type: actionTypesAccount.SET_ACCOUNT_STATUS,
          payload: {
            id: statusNames.UPDATE_AWS_FIELDS,
            status: requestStatusTypes.LOADING,
          },
        },
        {
          type: actionTypesAccount.SET_ACCOUNT_STATUS,
          payload: {
            id: statusNames.UPDATE_AWS_FIELDS,
            status: requestStatusTypes.SUCCESS,
          },
        },
      ];

      Auth.updateUserAttributes.mockReturnValue(Promise.resolve());

      const result = await store.dispatch(requestUpdateAWSFields(updatePayload));

      const actions = store.getActions();

      expect(actions).toEqual(expectedActions);
      expect(result).toEqual(null);
      expect(requestAWSUser).not.toBeCalled();
      expect(Auth.updateUserAttributes).toBeCalledWith(userObject.user, expectedUpdatePayload);
      expect(requestLogout).not.toBeCalled();
    });
  });
});
