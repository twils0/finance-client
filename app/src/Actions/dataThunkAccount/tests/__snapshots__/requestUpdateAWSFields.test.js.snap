// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dataThunkAccount requestUpdateAWSFields fails and throws an error when missing 'name', 'email', and 'phone_number' keys in payload 1`] = `[Error: Please enter a value for either the 'email', 'name', or 'phone' keys- {}]`;
